src/main/java/mezz/jei/plugins/vanilla/anvil/AnvilRecipeMaker.java

```diff
@@ -0,0 +1,127 @@
+import mezz.jei.api.IModRegistry;
+import mezz.jei.util.Log;
+import net.minecraft.enchantment.Enchantment;
+import net.minecraft.enchantment.EnchantmentHelper;
+import net.minecraft.init.Blocks;
+import net.minecraft.init.Items;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.fml.common.registry.ForgeRegistries;
+import net.minecraftforge.oredict.OreDictionary;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public class AnvilRecipeMaker {
+
+	public static void registerVanillaAnvilRecipes(IModRegistry registry) {
+		Stopwatch sw = Stopwatch.createStarted();
+		registerRepairRecipes(registry);
+		sw.stop();
+		Log.info("Registered vanilla repair recipes in {} ms", sw.elapsed(TimeUnit.MILLISECONDS));
+		sw.reset();
+		sw.start();
+		registerBookEnchantmentRecipes(registry);
+		sw.stop();
+		Log.info("Registered enchantment recipes in {} ms", sw.elapsed(TimeUnit.MILLISECONDS));
+	}
+
+	private static void registerBookEnchantmentRecipes(IModRegistry registry) {
+		List<ItemStack> ingredients = registry.getIngredientRegistry().getIngredients(ItemStack.class);
+		List<Enchantment> enchantments = ForgeRegistries.ENCHANTMENTS.getValues();
+		ItemStack book = new ItemStack(Items.ENCHANTED_BOOK);
+		for (ItemStack ingredient : ingredients) {
+			for (Enchantment enchantment : enchantments) {
+				if (enchantment.canApply(ingredient)) {
+					List<ItemStack> perLevelBooks = Lists.newArrayList();
+					List<ItemStack> perLevelOutputs = Lists.newArrayList();
+					for (int level = 1; level <= enchantment.getMaxLevel(); level++) {
+						ItemStack withEnchant = ingredient.copy();
+						ItemStack bookEnchant = book.copy();
+						Map<Enchantment, Integer> enchMap = Collections.singletonMap(enchantment, level);
+						EnchantmentHelper.setEnchantments(enchMap, withEnchant);
+						EnchantmentHelper.setEnchantments(enchMap, bookEnchant);
+						perLevelBooks.add(bookEnchant);
+						perLevelOutputs.add(withEnchant);
+					}
+					registry.addAnvilRecipe(ingredient, perLevelBooks, perLevelOutputs);
+				}
+			}
+		}
+	}
+
+	@SuppressWarnings("unchecked")
+	private static void registerRepairRecipes(IModRegistry registry) {
+		ItemStack repairWood = new ItemStack(Blocks.PLANKS, 1, OreDictionary.WILDCARD_VALUE);
+		ItemStack repairStone = new ItemStack(Blocks.COBBLESTONE);
+		ItemStack repairIron = new ItemStack(Items.IRON_INGOT);
+		ItemStack repairGold = new ItemStack(Items.GOLD_INGOT);
+		ItemStack repairDiamond = new ItemStack(Items.DIAMOND);
+		ItemStack repairLeather = new ItemStack(Items.LEATHER);
+		List<Pair<ItemStack,ItemStack>> items = Lists.newArrayList(
+				Pair.of(new ItemStack(Items.WOODEN_SWORD), repairWood),
+				Pair.of(new ItemStack(Items.WOODEN_PICKAXE), repairWood),
+				Pair.of(new ItemStack(Items.WOODEN_AXE), repairWood),
+				Pair.of(new ItemStack(Items.WOODEN_SHOVEL), repairWood),
```

Comment:
I think this would be cleaner as a `Map`:
```java
Map<ItemStack, List<ItemStack>> items = new HashMap<>();
items.put(repairWood, Arrays.asList(
    new ItemStack(Items.WOODEN_SWORD),
    new ItemStack(Items.WOODEN_PICKAXE),
    ...
));
```